From f7cb76b83af63ee55b7654883619f93df0fa3089 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?V=C3=A1clav=20Pavl=C3=ADn?= <vpavlin@redhat.com>
Date: Mon, 29 Jul 2013 14:09:45 +0200
Subject: [PATCH 5/7] Remove grub.info when make clean is called

Resolves: rhbz#854652
---
 docs/Makefile.am | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/docs/Makefile.am b/docs/Makefile.am
index fe6b22b..4d5fc31 100644
--- a/docs/Makefile.am
+++ b/docs/Makefile.am
@@ -8,6 +8,7 @@ HELP2MAN = help2man
 SRC2TEXI = src2texi
 noinst_SCRIPTS = $(HELP2MAN) $(SRC2TEXI)
 EXTRA_PROGRAMS = kernel
+man_INFOS = grub.info
 
 # The example kernel is built if you specify --enable-example-kernel.
 if BUILD_EXAMPLE_KERNEL
@@ -22,7 +23,7 @@ endif
 
 EXTRA_DIST = menu.lst $(man_MANS) $(noinst_SCRIPTS) \
 	$(EXAMPLES) $(multiboot_TEXINFOS)
-CLEANFILES = $(noinst_PROGRAMS)
+CLEANFILES = $(noinst_PROGRAMS) $(man_INFOS)
 
 # Cancel the rule %.texi -> %. This rule may confuse make to determine
 # the dependecies.
-- 
1.8.3.1

