From 0f603a41a93f27a569a0d9918507d6378bfea07a Mon Sep 17 00:00:00 2001
From: Laszlo Ersek <lersek@redhat.com>
Date: Wed, 27 Feb 2013 01:59:26 +0100
Subject: [PATCH 7/7] Recognize virtio-blk devices when running on OVMF

Signed-off-by: Laszlo Ersek <lersek@redhat.com>
---
 efi/efidisk.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/efi/efidisk.c b/efi/efidisk.c
index f46f24b..001ad89 100644
--- a/efi/efidisk.c
+++ b/efi/efidisk.c
@@ -204,7 +204,9 @@ name_devices (struct grub_efidisk_data *devices)
 	      add_device (&hd_devices, d);
 	    }
 	}
-      else if (GRUB_EFI_DEVICE_PATH_TYPE(dp) == GRUB_EFI_HARDWARE_DEVICE_PATH_TYPE && GRUB_EFI_DEVICE_PATH_SUBTYPE(dp) == GRUB_EFI_VENDOR_DEVICE_PATH_SUBTYPE)
+      else if (GRUB_EFI_DEVICE_PATH_TYPE(dp) == GRUB_EFI_HARDWARE_DEVICE_PATH_TYPE
+               && (GRUB_EFI_DEVICE_PATH_SUBTYPE(dp) == GRUB_EFI_VENDOR_DEVICE_PATH_SUBTYPE
+                   || GRUB_EFI_DEVICE_PATH_SUBTYPE(dp) == GRUB_EFI_PCI_DEVICE_PATH_SUBTYPE))
 	{
 	  add_device (&hd_devices, d);
 	}
-- 
1.8.3.1

