diff --git a/util/grub-install.in b/util/grub-install.in
index bbf3b5e..d4a5262 100644
--- a/util/grub-install.in
+++ b/util/grub-install.in
@@ -110,6 +110,8 @@ convert () {
 	  -e 's%\(/mapper/[[:alpha:]]\+[[:digit:]]*\)p[[:digit:]]\+$%\1%' \
 	  -e 's%\(/mapper/isw_[[:alpha:]_]\+[[:alpha:]]\+[[:digit:]]\+\)p[[:digit:]]\+$%\1%' \
 	  -e 's%\(/mapper/[[:alpha:]]\+_[[:alpha:]]\+\)[[:digit:]]\+$%\1%' \
+	  -e 's%\(/mapper/[[:alnum:]]\+\(_[[:alnum:]]\+\)\+\)\p[[:digit:]]\+$%\1%' \
+	  -e 's%\(/mapper/[[:alnum:]_-]\+\)\+p[[:digit:]]\+$%\1%' \
           -e 's%\(nvme[0-9]*n[0-9]*\)p[0-9]*$%\1%'`
 	tmp_part=`echo "$1" | grep -v '/mapper/control$' |
 		grep -v '/mapper/[[:alnum:]]\+-[[:alnum:]]\+$' | uniq |
@@ -122,6 +124,8 @@ convert () {
 	  -e 's%.*/mapper/[[:alpha:]]\+[[:digit:]]*p\([[:digit:]]\+\)$%\1%' \
 	  -e 's%.*/mapper/isw_[[:alpha:]_]\+[[:alpha:]]\+[[:digit:]]\+p\([[:digit:]]\+\)$%\1%' \
 	  -e 's%.*/mapper/[[:alpha:]]\+_[[:alpha:]]\+\([[:digit:]]\+\)$%\1%' \
+	  -e 's%.*/mapper/[[:alnum:]]\+\(_[[:alpha:]]\+[[:digit:]]*\)\+p\([[:digit:]]\+\)$%\2%' \
+	  -e 's%.*/mapper/[[:alnum:]_-]\+p\([[:digit:]]\+\)$%\1%' \
           -e 's%.*/nvme[0-9]*n[0-9]*p\([0-9]*\)%\1%' |
 		grep -v '.*/mapper/.*'`
 	;;
