diff --git a/efi/pxe.c b/efi/pxe.c
index 0a29c47..d23a5de 100644
--- a/efi/pxe.c
+++ b/efi/pxe.c
@@ -215,12 +215,13 @@ stra2ip6(char *str)
         }
         len = strlen((char *)str);
         a = b = (grub_efi_uint8_t *)str;
-        for(i=p=0; i < len; i++, b++) {
-                if (*b != ':')
+        for(i=p=0; i <= len; i++, b++) {
+                if (*b != ':' && *b != '\0')
                         continue;
                 *b = '\0';
                 ip[p++] = stra2ns(a);
-                *b = ':';
+                if (i != len)
+                        *b = ':';
                 a = b + 1;
                 if ( *(b+1) == ':' )
                         break;
-- 

