From da46897d46c543c6786c9609666b2aa56b763299 Mon Sep 17 00:00:00 2001
From: "David Kaspar [Dee'Kej]" <dkaspar@redhat.com>
Date: Mon, 26 Sep 2016 14:52:05 +0200
Subject: [PATCH] grub-install: allow more than 1 letter in the device name

---
 util/grub-install.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/util/grub-install.in b/util/grub-install.in
index d4a5262..1ecd70d 100644
--- a/util/grub-install.in
+++ b/util/grub-install.in
@@ -102,7 +102,7 @@ convert () {
     linux*)
 	tmp_disk=`echo "$1" | grep -v '/mapper/control$' |
 		grep -v '/mapper/[[:alnum:]]\+-[[:alnum:]]\+$' | uniq |
-		sed -e 's%\([shv]d[a-z]\)[0-9]*$%\1%' \
+		sed -e 's%\([shv]d[[:lower:]]\+\)[0-9]*$%\1%' \
 				  -e 's%\(d[0-9]*\)p[0-9]*$%\1%' \
 				  -e 's%\(fd[0-9]*\)$%\1%' \
 				  -e 's%/part[0-9]*$%/disc%' \
@@ -115,7 +115,7 @@ convert () {
           -e 's%\(nvme[0-9]*n[0-9]*\)p[0-9]*$%\1%'`
 	tmp_part=`echo "$1" | grep -v '/mapper/control$' |
 		grep -v '/mapper/[[:alnum:]]\+-[[:alnum:]]\+$' | uniq |
-		sed -e 's%.*/[shv]d[a-z]\([0-9]*\)$%\1%' \
+		sed -e 's%.*/[shv]d[[:lower:]]\+\([0-9]*\)$%\1%' \
 				  -e 's%.*d[0-9]*p%%' \
 				  -e 's%.*/fd[0-9]*$%%' \
 				  -e 's%.*/floppy/[0-9]*$%%' \
-- 
2.7.4

