From e2f1c9f779c3fe8049a053dee10bd9cd3ca62667 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Thu, 13 Oct 2011 16:35:36 -0400
Subject: [PATCH] Accept vendor media paths...

---
 efi/efidisk.c |   20 ++++++++++++++++++++
 1 files changed, 20 insertions(+), 0 deletions(-)

diff --git a/efi/efidisk.c b/efi/efidisk.c
index 6805413..f196f06 100644
--- a/efi/efidisk.c
+++ b/efi/efidisk.c
@@ -204,6 +204,10 @@ name_devices (struct grub_efidisk_data *devices)
 	      add_device (&hd_devices, d);
 	    }
 	}
+      else if (GRUB_EFI_DEVICE_PATH_TYPE(dp) == GRUB_EFI_HARDWARE_DEVICE_PATH_TYPE && GRUB_EFI_DEVICE_PATH_SUBTYPE(dp) == GRUB_EFI_VENDOR_DEVICE_PATH_SUBTYPE)
+	{
+	  add_device (&hd_devices, d);
+	}
       if (GRUB_EFI_DEVICE_PATH_TYPE(dp) == GRUB_EFI_ACPI_DEVICE_PATH_TYPE)
 	{
 	  add_device (&fd_devices, d);
-- 
1.7.6

