From 96068d8d3f06102faaa03d9fe153fca98a1c4099 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?V=C3=A1clav=20Pavl=C3=ADn?= <vpavlin@redhat.com>
Date: Thu, 6 Dec 2012 15:10:58 +0100
Subject: [PATCH] Check if pxe->Mode is not NULL

---
 efi/pxe.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/efi/pxe.c b/efi/pxe.c
index ef6311a..3503d46 100644
--- a/efi/pxe.c
+++ b/efi/pxe.c
@@ -723,7 +723,7 @@ char *grub_efi_pxe_get_config_path(grub_efi_loaded_image_t *LoadedImage)
 	}
 	grub_free(handles);
 
-	if (!pxe)
+	if (!pxe || !pxe->Mode)
 		return NULL;
 
 	if (pxe->Mode->UsingIpv6) {
-- 
1.7.11.7

