From dfc91ea5f18bb76db3a942b762f2c69a16aa901a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?V=C3=A1clav=20Pavl=C3=ADn?= <vpavlin@redhat.com>
Date: Tue, 9 Oct 2012 09:59:40 +0200
Subject: [PATCH 2/3] bz#737732 Let user specify efi boot disk as root

---
 stage2/disk_io.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/stage2/disk_io.c b/stage2/disk_io.c
index 46a46f7..1653e9f 100644
--- a/stage2/disk_io.c
+++ b/stage2/disk_io.c
@@ -1088,7 +1088,8 @@ set_device (char *device)
 #if defined(SUPPORT_NETBOOT) || defined(PLATFORM_EFI)
 	  if (*device == 'f' || *device == 'h'
 	      || (*device == 'n' && network_ready)
-	      || (*device == 'c' && cdrom_drive != GRUB_INVALID_DRIVE))
+	      || (*device == 'c' && cdrom_drive != GRUB_INVALID_DRIVE)
+	      || (*device == 'b'))
 #else
 	  if (*device == 'f' || *device == 'h'
 	      || (*device == 'c' && cdrom_drive != GRUB_INVALID_DRIVE))
@@ -1111,6 +1112,7 @@ set_device (char *device)
 	       || *device == 'h'
 #if defined(SUPPORT_NETBOOT) || defined(PLATFORM_EFI)
 	       || (*device == 'n' && network_ready)
+	       || (*device == 'b')
 #endif
 	       || (*device == 'c' && cdrom_drive != GRUB_INVALID_DRIVE))
 	      && (device += 2, (*(device - 1) != 'd')))
@@ -1124,6 +1126,8 @@ set_device (char *device)
 	    {
 	      if (ch == 'c' && cdrom_drive != GRUB_INVALID_DRIVE)
 		current_drive = cdrom_drive;
+	      else if (ch == 'b')
+		current_drive = boot_drive;
 	      else
 		{
 		  safe_parse_maxint (&device, (int *) &current_drive);
-- 
1.7.11.4

